app/src/main/java/com/ubergeek42/WeechatAndroid/service/P.java

```diff
@@ -125,7 +125,30 @@ public static void storeThemeOrColorSchemeColors(Context context) {
 
     public static boolean showSend, showTab, showPaperclip, hotlistSync;
 
-    public enum VolumeRole {NONE, TEXT_SIZE, SEND_HISTORY};
+    public enum VolumeRole {
+        DoNothing("none"),
+        ChangeTextSize("change_text_size"),
+        NavigateInputHistory("navigate_input_history");
+
+        public final String value;
+
+        VolumeRole(String value) {
+            this.value = value;
+        }
+
+        static VolumeRole valueOfName(String value) {
```

Comment:
Again, what is `value of name`?... What `change_text_size` is to `ChangeTextSize` is ”a string representation of enum value” (for the purpose of storage). (CMIIW: `VolumeRole` is “an enum” and `ChangeTextSize` is “an enum value”.) I'd use `stringRepresentation`, or just `string` for shirt and because it pairs nicely with factory methods `from*`.

And this *is* a kind of a factory methods and I prefer naming them all `from*`, here it would be `fromStringRepresentation()` or `fromStoredString()` or `fromPersistedString()` or just `fromString()`, like here:

https://github.com/ubergeek42/weechat-android/blob/822ceb70366d39b9d027ed6cf0cf059fd19081b8/relay/src/main/java/com/ubergeek42/weechat/relay/connection/Handshake.kt#L121-L137

Reply:
Fixed.